Update Licensing FAQs for v24.3 release (#19002)
cockroachdb/docs: src/current/_includes/common/licensing-faqs.md

CockroachDB is made available under the [CockroachDB Software License][csl].

## Types of licenses

The table of licenses below refers to options for {{ site.data.products.core }} deployments. All {{ site.data.products.cloud }} deployments automatically have a valid **Enterprise** license.

Type | Description
-------------|------------
**Enterprise** <a name="enterprise"></a> | This paid license allows usage of all CockroachDB features with no restrictions under the terms specified in the [CockroachDB Software License][csl]. License must be renewed annually or as negotiated. Support levels available include [Enterprise][support] or [Essential][support].

> **Enterprise Free** <a name="enterprise-free"></a> | Same functionality as **Enterprise**, but free of charge for businesses with less than $10M in annual revenue, and telemetry is required except for ephemeral clusters (7 days or less). Clusters will be [throttled](#throttling) after 7 days without sending telemetry. License must be renewed annually. Support level available is Community (i.e., [Docs]({% link {{ page.version.version }}/index.md %}), [Forum][forum], [Slack][slack]).

Reviewer: I feel like the "ephemeral" language here is confusing. I would delete ", and telemetry is required except for ephemeral clusters (7 days or less)".

Reviewer: why do we mention the "7 days without sending telemetry" for the Enterprise Free license and not the Enterprise Trial license? I think these should be the same in terms of telemetry.

Author: > I feel like the "ephemeral" language here is confusing. I would delete ", and telemetry is required except for ephemeral clusters (7 days or less)".

ok will remove it

> why do we mention the "7 days without sending telemetry" for the Enterprise Free license and not the Enterprise Trial license? I think these should be the same in terms of telemetry.

no reason, human error.  Updated as shown below (removed ephemeral bit from Free, added the 7 days to Trial):

**Enterprise Free** <a name="enterprise-free"></a> | Same functionality as **Enterprise**, but free of charge for businesses with less than $10M in annual revenue. Clusters will be [throttled](#throttling) after 7 days without sending telemetry. License must be renewed annually. Support level available is Community (i.e., [Docs]({% link {{ page.version.version }}/index.md %}), [Forum][forum], [Slack][slack]).
**Enterprise Trial** <a name="enterprise-trial"></a> | A 30 day self-service trial license. Telemetry is required during the trial. Clusters will be [throttled](#throttling) after 7 days without sending telemetry. Telemetry can be disabled once the cluster is upgraded to a paid **Enterprise** license. Support level available during trials is Community (i.e., [Docs]({% link {{ page.version.version }}/index.md %}), [Forum][forum], [Slack][slack]).

**Enterprise Trial** <a name="enterprise-trial"></a> | A 30 day self-service trial license. Telemetry is required during the trial. Telemetry can be disabled once the cluster is upgraded to a paid **Enterprise** license. Support level available during trials is Community (i.e., [Docs]({% link {{ page.version.version }}/index.md %}), [Forum][forum], [Slack][slack]).

{{site.data.alerts.callout_success}}
Note that:

> - Clusters with no license have a 7-day grace period before needing to install a license and start sending telemetry data.

Reviewer: I would remove the "and start sending telemetry data" part. New clusters can use any license key, doesn't have to be a telemetry license.

Reviewer: You could also mention that this is useful for ephemeral development clusters.

Reviewer: actually - we should think about how to phrase this in docs but: 

Self-hosted clusters without a license key are restricted to development purposes only. Development purposes includes non-production use in an internal development environment for design, prototype, or development purposes only. This restriction is enforced through license terms and conditions.

Author: updated this section a bit based on your feedback: (1) remove the "and start sending ..."; (2) mentioned the usefulness for dev clusters; (3) added the unlicensed dev cluster restriction info as an addendum at bottom of section just to make it super clear, also re-ordering the bullets to try to make more sense in context :

---

Note that:
- Clusters with Enterprise Free or Enterprise Trial licenses cannot disable telemetry; if such a cluster signals a telemetry sending error, it will be due to firewall configuration or a network issue.
- Clusters with no license have a 7-day grace period before needing to install a license. This is useful for ephemeral development clusters.
- No license key is required for developers running [single-node clusters](#single-node-clusters).

In general, self-hosted clusters without a license key are restricted to development purposes only. Development purposes includes non-production use in an internal development environment for design, prototype, or development purposes only. This restriction is enforced through [license terms and conditions][csl].

- No license key is required for developers running [single-node clusters](#single-node-clusters).

> - Clusters with Enterprise Free or Enterprise Trial licenses cannot disable telemetry; if such a cluster signals a telemetry sending error, it will be due to firewall configuration or a network issue.

Reviewer 2: LGTM

{{site.data.alerts.end}}

## Obtain a license

To obtain a paid **Enterprise** license, <a href="mailto:sales@cockroachlabs.com">contact Sales</a>.

To obtain an **Enterprise Free** or **Enterprise Trial** license, take the following steps:

1. Point your web browser to the [CockroachDB {{ site.data.products.cloud }} Console](https://cockroachlabs.cloud) and [log in]({% link cockroachcloud/authentication.md  %}#console-ui-authentication).

> 1. Select **Organization &raquo; Enterprise Licenses** from the nav bar dropdown. This will bring you to the **Enterprise Licenses** page, which shows a (possibly empty) list of licenses and their keys, with information about the status of each.

Reviewer: we should mention that you must have "Organization Admin" permissions

Author: updated the first step to 

"... log in as an account with **Organization Admin** permissions."

1. Click the **Create License** button. This will bring you to a page called **Get started with CockroachDB Enterprise**.
1. On this page, you will create an **Enterprise Trial** license or an **Enterprise Free** license.
    1. To create an **Enterprise Trial** license:
        1. Fill in the form with the required information.
        1. Click the **Continue** button.
        1. Select the checkbox to agree to the Terms & Conditions of the [CockroachDB Software License][csl].
        1. Click the **Generate License Key** button. You will be redirected to the **Enterprise Licenses** page, where you can [start using the key](#set-a-license).
    1. To create an **Enterprise Free** license:
        1. Fill in the form with the required information.

>         1. Toggle the switch called **Find out if my company qualifies for an Enterprise Free license**. *By toggling this switch you are legally attesting to the fact that your company revenue meets the requirements of the license*.

Reviewer: I think we should have another step after toggle the switch saying "Fill out the additional information". I think the disclaimer "By toggling..." should be moved to this next step.

Author: Done, PTAL

>         1. Click the **Continue** button.

Reviewer: At this stage, they may not qualify for the Free license. If they don't qualify, we will give them a Enterprise Trial license instead.

Author: added a bullet below this step that says

" - Note that at this stage, you may not qualify for the **Enterprise Free** license. If so, you will be given an **Enterprise Trial** license."

        1. Select the checkbox to agree to the Terms & Conditions of the [CockroachDB Software License][csl].
        1. Click the **Generate License Key** button. You will be redirected to the **Enterprise Licenses** page, where you can [start using the key](#set-a-license).

{{site.data.alerts.callout_danger}}
You will not be able to create more than one **Enterprise Trial** license per day. If you try, the UI will prevent you from proceeding, and the following message will be displayed:

> > A new Enterprise Trial license cannot be created at this time because one was recently created. Tell us more about your intended use to find out if you qualify for an Enterprise Free license, or create a new one after October 16, 2024 at 10:49 AM EDT.

Reviewer: This mentions an actual date and time.

Author: thanks, fixed!

{{site.data.alerts.end}}

{{site.data.alerts.callout_success}}
For quick local testing of Enterprise features, you can [run a single-node cluster](#single-node-clusters).
{{site.data.alerts.end}}

## Set a license

> {% include common/set-enterprise-license.md %}

Reviewer: Should we add a disclaimer here that you cannot apply a Enterprise Trial license more than once to a cluster?

Author: I added a note to the bottom of the included instructions that 

"You cannot apply an **Enterprise Trial** license more than once to the same cluster."

figured i'd keep it short b/c we document elsewhere the UI message re: not giving out multiple Trial licenses, pls contact Sales etc - let me know if that works!

{% include_cached copy-clipboard.html %}
~~~ sql
>  SHOW CLUSTER SETTING enterprise.license;
~~~
~~~
             enterprise.license
+-------------------------------------------+
  crl-0-ChB1x...
(1 row)
~~~

The license setting is also logged in the cockroach.log on the node where the command is run:

{% include_cached copy-clipboard.html %}
~~~ sql
$ cat cockroach.log | grep license
~~~
~~~
I171116 18:11:48.279604 1514 sql/event_log.go:102  [client=[::1]:56357,user=root,n1] Event: "set_cluster_setting", target: 0, info: {SettingName:enterprise.license Value:xxxxxxxxxxxx User:root}
~~~

## Monitor for license expiry

You can monitor the time until your license expires in the following ways:

1. [Prometheus]({% link {{ page.version.version }}/monitor-cockroachdb-with-prometheus.md %}): The `seconds_until_enterprise_license_expiry` metric reports the number of seconds until the license on a cluster expires. It will report `0` if there is no license, and a negative number if the license has already expired. For more information, see [Monitoring and Alerting]({% link {{ page.version.version }}/monitoring-and-alerting.md %}).

> 1. [DB console]({% link {{ page.version.version }}/ui-overview.md %}): Several [license expiration messages]({% link {{ page.version.version }}/ui-overview.md %}#license-expiration-message) may be displayed, depending on the status of your cluster's license.

Reviewer: For Enterprise Free and Trial clusters, we also notify you in the CC console under the Enterprise Licenses section. We will also send emails accordingly.

Author: Nice, added another bullet:

1. [CockroachDB {{ site.data.products.cloud }} Console][cloud-console]: If you have an **Enterprise Free** or **Enterprise Trial** cluster, you will see notifications in the **Enterprise Licenses** section, as well as receive notification emails to the [email address associated with your account](#obtain-a-license).

Reviewer: great! the only additional comment is that emails are sent to the cloud org's admins (i.e., users with Org Admin permissions).

Author: modified to say the following in commit named `Emails go to those with Org Admin perms`

> ... as well as receive notification emails sent to users with **Org Admin** permissions.

PTAL!

## Renew an expired license

To renew an expired **Enterprise** license, <a href="mailto:sales@cockroachlabs.com">contact Sales</a> and then [set](#set-a-license) the new license.

To renew an expired **Enterprise Free** license, follow the steps in [Obtain a license](#obtain-a-license).

## FAQs

### Can I host CockroachDB as a service for internal use at my organization?

Yes, employees and contractors can use your internal CockroachDB instance as a service, but no people outside of your organization will be able to use it without purchasing a license. Use of Enterprise features will always require a license.

### What constitutes hosting CockroachDB as a service?

Hosting CockroachDB as a service means creating an offering that allows third parties (other than your employees and contractors) to operate a database. Specifically, third parties cannot modify table schemas.

<a name="throttling"></a>

### What is throttling and how does it work?

When a cluster is being throttled, the number of concurrent open [SQL transactions]({% link {{ page.version.version }}/transactions.md %}) is limited to 5.

> This will only happen in the following cases:

Reviewer: Throttling would also happen when you patch upgrade a core cluster (i.e., no license key and running a Core-supported version) to a Core-not-supported patch. These clusters get a 30 day grace period to de-risk impact to prod clusters.

Author: thanks this is a huge case - i hadn't seen this comment yet but this is why i changed "will only happen ..." to "will happen ..." :-) 

for this case, i added the following bullet:

- The cluster was running an earlier version of CockroachDB which supported the license known as "CockroachDB Core" and has just been patch upgraded to a version of CockroachDB which is available under the [CockroachDB Software License][csl]. Such clusters get a 30 day grace period before being throttled to de-risk impact to production environments.

- The cluster is not following telemetry requirements.

>     - There is a 7 day grace period for new **Enterprise Free** and **Enterprise Trial** clusters to start sending telemetry.

Reviewer: This shouldn't be "new" clusters only. Any time there is an interruption in telemetry, the grace period should start.

Author: Updated a bit to the following, PTAL
----
    - There is a 7 day grace period for **Enterprise Free** and **Enterprise Trial** clusters to (re)start sending telemetry. This applies when a cluster is newly created, or any time there is an interruption in telemetry.

- The cluster has an expired [license key](#obtain-a-license); depending on the type of expired license, the cluster will be throttled after the following time periods:
    - **Enterprise**: Never throttles
    - **Enterprise Free**: Throttles 30 days after expiration
    - **Enterprise Trial**: Throttles 7 days after expiration

>     - No license: Throttles 7 days after cluster initialization

Reviewer: I feel like this "No license" point doesn't make sense under this section on "expired license key". I think this should be broken out into its own point.

Author: Made this a separate standalone sentence after the bulleted list that says 

"If no valid license key is ever entered, the cluster will be throttled 7 days after cluster initialization."

also FYI I removed the text "only" from the top of this section, so it now says "this will happen in the following cases", which is still correct but leaves open the option to edit this text in future if/when someone internally or on the interwebs sees it later and goes "hey you missed one" - more of a good tech writing practice than anything to do with this PR specifically

Author: @jaiayu ok to resolve the convo based on this latest update?

Single node clusters for development use are [not throttled](#single-node-clusters).

### Can I use CockroachDB for academic research?

> Cockroach Labs encourages non-commercial academic research involving CockroachDB. For such projects, please [contact us][support] to discuss a possible licensing arrangement.

Reviewer: They can use Enterprise Free license for this purpose. No need to contact us.

Author: nice, updated

<a name="single-node-clusters"></a>

### Do I need a license key for running a single node cluster?

No license key is required for developers running [single-node clusters]({% link {{ page.version.version }}/cockroach-start-single-node.md %}) using the following [`cockroach` commands]({% link {{ page.version.version }}/cockroach-commands.md %}):

- [`cockroach start-single-node`]({% link {{ page.version.version }}/cockroach-start-single-node.md %})
- [`cockroach demo`]({% link {{ page.version.version }}/cockroach-demo.md %})

> Single node clusters are not [throttled](#throttling).

Reviewer: might be good to add that you should only be using clusters without a license key for development purposes (not prod).

Author: added the same new blurb as above to the end of this section:

In general, self-hosted clusters without a license key are restricted to development purposes only. Development purposes includes non-production use in an internal development environment for design, prototype, or development purposes only. This restriction is enforced through [license terms and conditions][csl].

CockroachDB is made available under the [CockroachDB Software License][csl].

## Types of licenses

The table of licenses below refers to options for {{ site.data.products.core }} deployments. All {{ site.data.products.cloud }} deployments automatically have a valid **Enterprise** license.

> Type | Description

Reviewer 3: We left out any mention of evaluation licenses. Was that intentional?

Author: what are "evaluation licenses" - is that something different than the self-serve Enterprise Trial license?  perhaps something that is handed out directly by sales?   I'm not seeing the text "evaluation" or "evaluation license" in the product brief

Reviewer 3: The evaluation license type existed previously (see 1.1.3 in the product brief). My understanding is that they are used for POCs.

Author: thanks @spilchen for the catch, i just reviewed that list again, it seems we are missing multiple items from that list, my bad

@jaiayu do you want me to replicate here all of the license types from the product brief?  specifically, it looks like we're missing:

- Evaluation
- Development
- Pre-production

Or is it our preference that those license types are described and managed via interactions with a Sales process vs. being described in the public user documentation? 

I'm asking because up until now we have not listed out such licenses - [current FAQ](https://www.cockroachlabs.com/docs/stable/licensing-faqs#types-of-licenses) says there is:

- Apache
- BSL
- CCL

And doesn't talk about dev, eval etc licenses.

Presumably dev, eval, etc. licenses also existed under the previous regime, but we never documented them before

Happy to do it now if you want, please let me know

Reviewer: @rmloveland I think it might be a good to mention that there is an option to evaluate CockroachDB without the telemetry requirements by contacting Sales. 

I'm not sure if we should list out the "Evaluation" license key, or if we should just say "Contact Sales if you want to try CockroachDB without telemetry requirements or if you require an extended trial period."

Author: Updated to add the following in a commit `Contact sales for longer/no-telemetry trials`

> <a href="mailto:sales@cockroachlabs.com">Contact Sales</a> if you want to try CockroachDB without telemetry requirements or if you require an extended trial period.

-------------|------------
**Enterprise** <a name="enterprise"></a> | This paid license allows usage of all CockroachDB features with no restrictions under the terms specified in the [CockroachDB Software License][csl]. License must be renewed annually or as negotiated. Support levels available include [Enterprise][support] or [Essential][support].

> **Enterprise Free** <a name="enterprise-free"></a> | Same functionality as **Enterprise**, but free of charge for businesses with less than $10M in annual revenue, and telemetry is required except for ephemeral clusters (7 days or less). Clusters will be [throttled](#throttling) after 7 days without sending telemetry. License must be renewed annually. Support level available is Community (i.e., [Docs]({% link {{ page.version.version }}/index.md %}), [Forum][forum], [Slack][slack]).

Reviewer 3: If the cluster is ephemeral, you wouldn't need to install a license at all. It might be helpful to explain the "no license" scenario, as users may want to know they have a 7-day grace period before needing to install a license and you don't have to send telemetry in the grace period.

Author: I updated the text at the bottom of this section to cover the no license and single-node dev cluster scenarios as follows. PTAL

> Note that:
> - Clusters with no license have a 7-day grace period before needing to install a license and start sending telemetry data.
> - No license key is required for developers running [single-node clusters](#single-node-clusters).

> **Enterprise Trial** <a name="enterprise-trial"></a> | A 30 day self-service trial license. Telemetry is required during the trial (with negotiated exceptions). Telemetry can be disabled once the cluster is upgraded to a paid **Enterprise** license. Support level available during trials is Community (i.e., [Docs]({% link {{ page.version.version }}/index.md %}), [Forum][forum], [Slack][slack]).

Author: @spilchen and @dhartunian - elsewhere you each commented that Enterprise Trial clusters must send telemetry - i also just reviewed the product brief - not sure why I wrote this "(with negotiated exceptions)" clause but I'm removing it FYI

~~~

The license setting is also logged in the cockroach.log on the node where the command is run:

{% include_cached copy-clipboard.html %}
~~~ sql
$ cat cockroach.log | grep license
~~~
~~~
I171116 18:11:48.279604 1514 sql/event_log.go:102  [client=[::1]:56357,user=root,n1] Event: "set_cluster_setting", target: 0, info: {SettingName:enterprise.license Value:xxxxxxxxxxxx User:root}
~~~

## Monitor for license expiry

You can monitor the time until your license expires in the following ways:

1. [Prometheus]({% link {{ page.version.version }}/monitor-cockroachdb-with-prometheus.md %}): The `seconds_until_enterprise_license_expiry` metric reports the number of seconds until the license on a cluster expires. It will report `0` if there is no license, and a negative number if the license has already expired. For more information, see [Monitoring and Alerting]({% link {{ page.version.version }}/monitoring-and-alerting.md %}).
1. [DB console]({% link {{ page.version.version }}/ui-overview.md %}): Several [license expiration messages]({% link {{ page.version.version }}/ui-overview.md %}#license-expiration-message) may be displayed, depending on the status of your cluster's license.

> ## Renew an expired license

Author: via @jaiayu :

> for older/current versions: Our new license types (Enterprise Trial and Enterprise Free) will work on older versions, but the expiration behavior will be different. Clusters won't be throttled, but the Enterprise features will immediately stop working at license expiration

@rmloveland make sure you update the FAQ with this, maybe here? maybe throttling section? TBD

Author: @jaiayu PS is that also true on a rolling basis going forward?

 eg when v25.1 drops will the "Enterprise Features immediately stop working at license expiration" behavior be applied backwards to e.g. v24.3.X as well as v24.2.X et al
 
or is it a one-time deal with the release of v24.3 and the new licensing scheme

Reviewer: I think the response is that it's a one time deal. 

Any binaries that already exist - enterprise features immediately stop working at license expiration
24.3+ and any new binaries of 23.1+ - Throttling behavior will apply.

Author: thanks for this callout 

I have added the following info to the 'Monitor for license expiry' section.  It's a bit awkward to describe all this logic in prose, but here's my attempt, PTAL

----

During the transition to the [CockroachDB Software License][csl], expiration behavior (including [throttling](#throttling)) will work as follows depending on your version of CockroachDB.

For versions v23.1.29, v23.2.15, v24.1.7, v24.2.5, and v24.3.0 and later, clusters will behave as described in this documentation.

For versions less than or equal to v23.1.28, v23.2.14, v24.1.6, and v24.2.4, the behavior will be as follows:

- **Enterprise Trial** and **Enterprise Free** licenses will work as expected when you [set a license](#set-a-license).
- However, the expiration behavior will be different. Clusters running these versions won't be [throttled](#throttling); instead, the Enterprise features will immediately stop working at license expiration.

To renew an expired **Enterprise** license, <a href="mailto:sales@cockroachlabs.com">contact Sales</a> and then [set](#set-a-license) the new license.

To renew an expired **Enterprise Free** license, follow the steps in [Obtain a license](#obtain-a-license).

## FAQs

### Can I host CockroachDB as a service for internal use at my organization?

Yes, employees and contractors can use your internal CockroachDB instance as a service, but no people outside of your organization will be able to use it without purchasing a license. Use of Enterprise features will always require a license.

### What constitutes hosting CockroachDB as a service?

Hosting CockroachDB as a service means creating an offering that allows third parties (other than your employees and contractors) to operate a database. Specifically, third parties cannot modify table schemas.

<a name="throttling"></a>

### What is throttling and how does it work?

> When a cluster is being throttled, the number of concurrent open [SQL transactions]({% link {{ page.version.version }}/transactions.md %}) limited to 5.

Reviewer 4: linked from [here](https://github.com/cockroachdb/docs/pull/19008#discussion_r1801851551), drive-by grammar correction:

```suggestion
When a cluster is being throttled, the number of concurrent open [SQL transactions]({% link {{ page.version.version }}/transactions.md %}) is limited to 5.
```

Author: ✅  thanks Florence!

This will only happen in the following cases:

> - The cluster has no [valid license key](#obtain-a-license).

Reviewer 3: Does "valid" simply mean the license doesn't exist? Throttling can also occur if you're operating with an expired license beyond the grace period:
- Free: throttles 30 days after expiration
- Trial: throttles 7 days after expiration
- None: throttles 7 days after cluster initialization
- Enterprise: never throttles

Author: thanks for laying this out explicitly, i have updated this section a bit to try to add the info and hopefully make things clearer - please let me know what you think

> This will only happen in the following cases:
> - The cluster is not following telemetry requirements.
>     - There is a 7 day grace period for new **Enterprise Free** and **Enterprise Trial** clusters to start sending telemetry.
> - The cluster has an expired [license key](#obtain-a-license); depending on the type of expired license, the cluster will be throttled after the following time periods:
>     - **Enterprise**: Never throttles
>     - **Enterprise Free**: Throttles 30 days after expiration
>     - **Enterprise Trial**: Throttles 7 days after expiration
>     - No license: Throttles 7 days after cluster initialization

- The cluster is not following telemetry requirements.

> There is a 7 day grace period for new **Enterprise Free** clusters to start sending telemetry.

Reviewer 3: Both Free and Trial need to send telemetry.

Author: Ah ok, i updated that sentence to the following, please let me know what you think:

> There is a 7 day grace period for new **Enterprise Free** and **Enterprise Trial** clusters to start sending telemetry.

{{site.data.alerts.callout_danger}}
You will not be able to create more than one **Enterprise Trial** license per day. If you try, the UI will prevent you from proceeding, and the following message will be displayed:

> A new Enterprise Trial license cannot be created at this time because one was recently created. Tell us more about your intended use to find out if you qualify for an Enterprise Free license, or create a new one after October 16, 2024 at 10:49 AM EDT.
{{site.data.alerts.end}}

{{site.data.alerts.callout_success}}
For quick local testing of Enterprise features, you can [run a single-node cluster](#single-node-clusters).
{{site.data.alerts.end}}

## Set a license

{% include common/set-enterprise-license.md %}

## Verify a license

To verify a license, open the [built-in SQL shell]({% link {{ page.version.version }}/cockroach-sql.md %}) and use the [`SHOW CLUSTER SETTING`]({% link {{ page.version.version }}/show-cluster-setting.md %}) command to check the organization name and license key:

{% include_cached copy-clipboard.html %}
~~~ sql

> >  SHOW CLUSTER SETTING cluster.organization;

Reviewer 3: Does this need to be mentioned? I don't think it matters now because we no longer cross check the organization in the license with this setting.

Author: Ah sorry, holdover from before.  Removed in latest commit `Update with spilchen feedback (2)`  - thanks!

1. Select **Organization &raquo; Enterprise Licenses** from the nav bar dropdown. This will bring you to the **Enterprise Licenses** page, which shows a (possibly empty) list of licenses and their keys, with information about the status of each.
1. Click the **Create License** button. This will bring you to a page called **Get started with CockroachDB Enterprise**.
1. On this page, you will create an **Enterprise Trial** license or an **Enterprise Free** license.
    1. To create an **Enterprise Trial** license:
        1. Fill in the form with the required information.
        1. Click the **Continue** button.
        1. Select the checkbox to agree to the Terms & Conditions of the [CockroachDB Software License][csl].
        1. Click the **Generate License Key** button. You will be redirected to the **Enterprise Licenses** page, where you can [start using the key](#set-a-license).
    1. To create an **Enterprise Free** license:
        1. Fill in the form with the required information.
        1. Toggle the switch called **Find out if my company qualifies for an Enterprise Free license**.

>         1. Fill out the additional information. *By toggling this switch you are legally attesting to the fact that your company revenue meets the requirements of the license*.

Reviewer: @rmloveland toggling does not legally attest that your company meets the requirements of the license. I think we should remove this line. 

They actually attest at Step E (when they select the T&Cs)

Author: removed it!

~~~

## Monitor for license expiry

You can monitor the time until your license expires in the following ways:

1. [Prometheus]({% link {{ page.version.version }}/monitor-cockroachdb-with-prometheus.md %}): The `seconds_until_enterprise_license_expiry` metric reports the number of seconds until the license on a cluster expires. It will report `0` if there is no license, and a negative number if the license has already expired. For more information, see [Monitoring and Alerting]({% link {{ page.version.version }}/monitoring-and-alerting.md %}).
1. [DB console]({% link {{ page.version.version }}/ui-overview.md %}): Several [license expiration messages]({% link {{ page.version.version }}/ui-overview.md %}#license-expiration-message) may be displayed, depending on the status of your cluster's license.
1. [CockroachDB {{ site.data.products.cloud }} Console][cloud-console]: If you have an **Enterprise Free** or **Enterprise Trial** cluster, you will see notifications in the **Enterprise Licenses** section, as well as receive notification emails to the [email address associated with your account](#obtain-a-license).

> 1. CockroachDB emits [log messages]({% {{page.version.version}}/logging-overview.md %}) when a cluster is at risk of being [throttled](#throttling) due to license expiration or telemetry requirements. The database will also return warnings or errors (as appropriate) to [SQL clients]({% link {{ page.version.version }}/cockroach-sql.md %}) or [applications]({% link {{ page.version.version }}/install-client-drivers.md %}) that try to execute [transactions]({% link {{ page.version.version }}/transactions.md %}).

Author: thanks @spilchen I've updated this (in commit `Update with spilchen feedback (3)`) to say "return notices" instead

PS thank you for the review!

{{site.data.alerts.callout_info}}
During the transition to the [CockroachDB Software License][csl], expiration behavior (including [throttling](#throttling)) will work as follows depending on your version of CockroachDB.

For versions v23.1.29, v23.2.15, v24.1.7, v24.2.5, and v24.3.0 and later, clusters will behave as described in this documentation.

For versions less than or equal to v23.1.28, v23.2.14, v24.1.6, and v24.2.4, the behavior will be as follows:

- **Enterprise Trial** and **Enterprise Free** licenses will work as expected when you [set a license](#set-a-license).
- However, the expiration behavior will be different. Clusters running these versions won't be [throttled](#throttling); instead, the Enterprise features will immediately stop working at license expiration.
{{site.data.alerts.end}}

## Renew an expired license

To renew an expired **Enterprise** license, <a href="mailto:sales@cockroachlabs.com">contact Sales</a> and then [set](#set-a-license) the new license.

To renew an expired **Enterprise Free** license, follow the steps in [Obtain a license](#obtain-a-license).

## FAQs

### Can I host CockroachDB as a service for internal use at my organization?

> Yes, employees and contractors can use your internal CockroachDB instance as a service, but no people outside of your organization will be able to use it without purchasing a license. Use of Enterprise features will always require a license.

Reviewer: Should we still say "use of Enterprise features" -- with the new licensing change, all features require a license.

Author: I deleted that sentence altogether

### What constitutes hosting CockroachDB as a service?

Hosting CockroachDB as a service means creating an offering that allows third parties (other than your employees and contractors) to operate a database. Specifically, third parties cannot modify table schemas.

<a name="throttling"></a>

### What is throttling and how does it work?

When a cluster is being throttled, the number of concurrent open [SQL transactions]({% link {{ page.version.version }}/transactions.md %}) is limited to 5.

This will happen in the following cases:

- The cluster is not following telemetry requirements.

>     - There is a 7 day grace period for **Enterprise Free** and **Enterprise Trial** clusters to (re)start sending telemetry. This applies when a cluster is newly created, or any time there is an interruption in telemetry.

Reviewer: I think this should say "applies when the license is first added" (or something along those lines) instead of when a cluster is newly created.

Author: changed to say "applies when the license is first added" and made "added" a link to the 'Set a license' instructions elsewhere on this page

> All versions of CockroachDB released after November 18, 2024, including patch fixes for prior versions 23.1 onward, are made available under the [CockroachDB Software License][csl]. You can use the dropdown menu at the top of the page to view documentation for other versions.

Reviewer 5: A little unclear what "prior versions 23.1 onward" means. Is the suggestion correct?

```suggestion
All versions of CockroachDB released after November 18, 2024, including patch fixes for v23.1 and later, are made available under the [CockroachDB Software License][csl]. You can use the dropdown menu at the top of the page to view documentation for other versions.
```

Author: this one got legal touches so even if it's a little unclear i would prefer to leave it be

<a name="types-of-licenses"></a>

## License options

The table of licenses below refers to options for {{ site.data.products.core }} deployments. All {{ site.data.products.cloud }} deployments automatically have a valid **Enterprise** license.

Type | Description
-------------|------------
**Enterprise** <a name="enterprise"></a> | This paid license allows usage of all CockroachDB features in accordance with the terms specified in the [CockroachDB Software License][csl]. License must be renewed annually or as negotiated. Support levels available include [Enterprise][support] or [Essential][support].
**Enterprise Free** <a name="enterprise-free"></a> | Same functionality as **Enterprise**, but free of charge for businesses with less than $10M in annual revenue. Clusters will be [throttled](#throttling) after 7 days without sending telemetry. License must be renewed annually. Support level available is Community (i.e., [Docs]({% link {{ page.version.version }}/index.md %}), [Forum][forum], [Slack][slack]).

> **Enterprise Trial** <a name="enterprise-trial"></a> | A 30 day self-service trial license. Telemetry is required during the trial. Clusters will be [throttled](#throttling) after 7 days without sending telemetry. Telemetry can be disabled once the cluster is upgraded to a paid **Enterprise** license. Support level available during trials is Community (i.e., [Docs]({% link {{ page.version.version }}/index.md %}), [Forum][forum], [Slack][slack]).

Reviewer 5: ```suggestion
**Enterprise Trial** <a name="enterprise-trial"></a> | A 30 day self-service trial license. Telemetry is required during the trial. Clusters will be [throttled](#throttling) after 7 days without sending telemetry. Telemetry can be disabled once the cluster is upgraded to a paid **Enterprise** license. Support level available during trial is Community (i.e., [Docs]({% link {{ page.version.version }}/index.md %}), [Forum][forum], [Slack][slack]).
```

Author: ✅

<a href="mailto:sales@cockroachlabs.com">Contact Sales</a> if you want to try CockroachDB without telemetry requirements or if you require an extended trial period.

{{site.data.alerts.callout_success}}
Note that:

> - Clusters with no license key have a 7-day grace period before requiring a license key. This is useful for ephemeral development clusters.

Reviewer 5: Need a line break above this; the bullets aren't rendering

Author: ARRRRRGGGGHHHHH MARKDOWNNNNNNNNNNN

✅

- No license key is required for developers running [single-node clusters](#single-node-clusters).
{{site.data.alerts.end}}

## Obtain a license

To obtain a paid **Enterprise** license, <a href="mailto:sales@cockroachlabs.com">contact Sales</a>.

To obtain an **Enterprise Free** or **Enterprise Trial** license, take the following steps:

> 1. Point your web browser to the [CockroachDB {{ site.data.products.cloud }} Console][cloud-console] and [log in]({% link cockroachcloud/authentication.md  %}#console-ui-authentication) as an account with **Organization Admin** permissions.

Reviewer 5: ```suggestion
1. Point your web browser to the [CockroachDB {{ site.data.products.cloud }} Console][cloud-console] and [log in]({% link cockroachcloud/authentication.md %}#console-ui-authentication) as an account with **Organization Admin** permissions.
```

Author: ✅

        1. Click the **Generate License Key** button. You will be redirected to the **Enterprise Licenses** page, where you can [start using the key](#set-a-license).
    1. To create an **Enterprise Free** license:
        1. Fill in the form with the required information.
        1. Toggle the switch called **Find out if my company qualifies for an Enterprise Free license**.
        1. Fill out the additional information.
        1. Click the **Continue** button.
           - Note that at this stage, you may not qualify for the **Enterprise Free** license. If so, you will be given an **Enterprise Trial** license.
        1. Select the checkbox to agree to the Terms & Conditions of the [CockroachDB Software License][csl].
        1. Click the **Generate License Key** button. You will be redirected to the **Enterprise Licenses** page, where you can [start using the key](#set-a-license).

{{site.data.alerts.callout_danger}}
You will not be able to create more than one **Enterprise Trial** license per day. If you try, the UI will prevent you from proceeding, and the following message will be displayed:

> > A new Enterprise Trial license cannot be created at this time because one was recently created. Tell us more about your intended use to find out if you qualify for an Enterprise Free license, or create a new one after ${DATE}.

Reviewer 5: This block quote isn't working within the callout, so it reads like normal documentation. Maybe italicize it instead?

Author: went with italics

refer to:  ARRRRRGGHHHHHHH MARKDOWNNNNNNNNNNNNNNNN

{{site.data.alerts.end}}

{{site.data.alerts.callout_success}}
For quick local testing of Enterprise features, you can [run a single-node cluster](#single-node-clusters).
{{site.data.alerts.end}}

## Set a license

{% include common/set-enterprise-license.md %}

## Verify a license

To verify a license, open the [built-in SQL shell]({% link {{ page.version.version }}/cockroach-sql.md %}) and use the [`SHOW CLUSTER SETTING`]({% link {{ page.version.version }}/show-cluster-setting.md %}) command to check the license key:

{% include_cached copy-clipboard.html %}
~~~ sql

> >  SHOW CLUSTER SETTING enterprise.license;

Reviewer 5: This shouldn't be necessary anymore.

```suggestion
SHOW CLUSTER SETTING enterprise.license;
```

Author: ah yes that's from the before times

✅

~~~
~~~
             enterprise.license
+-------------------------------------------+
  crl-0-ChB1x...
(1 row)
~~~

> The license setting is also logged in the cockroach.log on the node where the command is run:

Reviewer 5: Perhaps link to https://www.cockroachlabs.com/docs/v24.3/logging-overview.html#logging-destinations

```suggestion
The license setting is also logged in the `cockroach.log` on the node where the command is run:
```

Author: ✅

{% include_cached copy-clipboard.html %}
~~~ sql
$ cat cockroach.log | grep license
~~~
~~~
I171116 18:11:48.279604 1514 sql/event_log.go:102  [client=[::1]:56357,user=root,n1] Event: "set_cluster_setting", target: 0, info: {SettingName:enterprise.license Value:xxxxxxxxxxxx User:root}
~~~

## Monitor for license expiry

You can monitor the time until your license expires in the following ways:

1. [Prometheus]({% link {{ page.version.version }}/monitor-cockroachdb-with-prometheus.md %}): The `seconds_until_enterprise_license_expiry` metric reports the number of seconds until the license on a cluster expires. It will report `0` if there is no license, and a negative number if the license has already expired. For more information, see [Monitoring and Alerting]({% link {{ page.version.version }}/monitoring-and-alerting.md %}).

> 1. [DB console]({% link {{ page.version.version }}/ui-overview.md %}): Several [license expiration messages]({% link {{ page.version.version }}/ui-overview.md %}#license-expiration-message) may be displayed, depending on the status of your cluster's license.

Reviewer 5: ```suggestion
1. [DB Console]({% link {{ page.version.version }}/ui-overview.md %}): Several [license expiration messages]({% link {{ page.version.version }}/ui-overview.md %}#license-expiration-message) may be displayed, depending on the status of your cluster's license.
```

Author: ✅

> 1. [CockroachDB {{ site.data.products.cloud }} Console][cloud-console]: If you have an **Enterprise Free** or **Enterprise Trial** cluster, you will see notifications in the **Enterprise Licenses** section, as well as receive notification emails sent to users with **Org Admin** permissions.

Reviewer 5: I forgot to make this correction!

```suggestion
1. [CockroachDB {{ site.data.products.cloud }} Console][cloud-console]: If you have an **Enterprise Free** or **Enterprise Trial** cluster, you will see notifications in the **Enterprise Licenses** section, as well as receive notification emails sent to users with **Organization Admin** permissions.
```

Author: ✅

1. CockroachDB emits [log messages]({% link {{ page.version.version }}/logging-overview.md %}) when a cluster is at risk of being [throttled](#throttling) due to license expiration or telemetry requirements. The database will also return notices to [SQL clients]({% link {{ page.version.version }}/cockroach-sql.md %}) or [applications]({% link {{ page.version.version }}/install-client-drivers.md %}) that try to execute [transactions]({% link {{ page.version.version }}/transactions.md %}).

{{site.data.alerts.callout_info}}
During the transition to the [CockroachDB Software License][csl], expiration behavior (including [throttling](#throttling)) will work as follows depending on your version of CockroachDB.

> For versions v23.1.29, v23.2.15, v24.1.7, v24.2.5, and v24.3.0 and later, clusters will behave as described in this documentation.

Author: ✅

> For versions less than or equal to v23.1.28, v23.2.14, v24.1.6, and v24.2.4, the behavior will be as follows:

Author: this one i'm feeling a little squirrely about due to the legal review already having happened, will leave it there for now 

for example i'd hate for "and earlier" to be construed as only referring to v24.2.4 instead of what i know you obv. mean which is "x.xx and earlier, y.yy and earlier, z.zz and earlier", etc.

but IANAL so i'm gonna leave this one

{{site.data.alerts.end}}

## Renew an expired license

To renew an expired **Enterprise** license, <a href="mailto:sales@cockroachlabs.com">contact Sales</a> and then [set](#set-a-license) the new license.

To renew an expired **Enterprise Free** license, follow the steps in [Obtain a license](#obtain-a-license).

## FAQs

### Can I host CockroachDB as a service for internal use at my organization?

Yes, employees and contractors can use your internal CockroachDB instance as a service, but no people outside of your organization will be able to use it without purchasing a license.

### What constitutes hosting CockroachDB as a service?

Hosting CockroachDB as a service means creating an offering that allows third parties (other than your employees and contractors) to operate a database. Specifically, third parties cannot modify table schemas.

<a name="throttling"></a>

### What is throttling and how does it work?

When a cluster is being throttled, the number of concurrent open [SQL transactions]({% link {{ page.version.version }}/transactions.md %}) is limited to 5.

This will happen in the following cases:

- The cluster is not following telemetry requirements.

>     - There is a 7 day grace period for **Enterprise Free** and **Enterprise Trial** clusters to (re)start sending telemetry. This applies when the license is first [added](#set-a-license), or any time there is an interruption in telemetry.

Author: i'm hesitant to touch this and the `30-day` one because these have been legally reviewed

paranoid? maybe 

ps i hope i'm not coming off as disagreeable or anything

- The cluster has an expired [license key](#obtain-a-license); depending on the type of expired license, the cluster will be throttled after the following time periods:
    - **Enterprise**: Never throttles
    - **Enterprise Free**: Throttles 30 days after expiration
    - **Enterprise Trial**: Throttles 7 days after expiration

> - The cluster was running an earlier version of CockroachDB which supported the license known as "CockroachDB Core" and has just been patch upgraded to a version of CockroachDB which is available under the [CockroachDB Software License][csl]. Such clusters get a 30 day grace period before being throttled to de-risk impact to production environments.

Author: same as above re: don't wanna touch this too much at this point

If no valid license key is ever entered, the cluster will be throttled 7 days after cluster initialization.

> Single node clusters for development use are [not throttled](#single-node-clusters).

Author: ✅

### Can I use CockroachDB for academic research?

Cockroach Labs encourages non-commercial academic research involving CockroachDB. For such projects, [obtain an **Enterprise Free** license](#obtain-a-license).

<a name="single-node-clusters"></a>

### Do I need a license key for running a single node cluster?

No license key is required for developers running [single-node clusters]({% link {{ page.version.version }}/cockroach-start-single-node.md %}) using the following [`cockroach` commands]({% link {{ page.version.version }}/cockroach-commands.md %}):

- [`cockroach start-single-node`]({% link {{ page.version.version }}/cockroach-start-single-node.md %})
- [`cockroach demo`]({% link {{ page.version.version }}/cockroach-demo.md %})

> Single node clusters are not [throttled](#throttling).

Author: ✅

> Generally single node clusters used for design, prototyping, testing, or development purposes in non-production, internal development environments do not require a license key.

Author: took the first two characters of this edit! tyty

CockroachDB is made available under the [CockroachDB Software License][csl].

## Types of licenses

> The table of licenses below refers to options for {{ site.data.products.core }} deployments. All {{ site.data.products.cloud }} deployments automatically have a valid **Enterprise** license.

Reviewer: @rmloveland what does the second line here mean?

Author: Basically if you use our cloud product your cloud-deployed binary already has an Enterprise license via your cloud situation so you don't have to worry / think about this

However it might not be necessary or even really make sense. Since it caught your eye I'm inclined to remove it because it may be saying "too much" about something that is not really necessary.

What do you think of the idea of removing it?
